### PR TITLE
[INV-35] Basic frontend layout

### PR DESCRIPTION
**Pull request by Aziza-Azizova**
## Description
- A Layout component is added.
- Header includes search and profile options for register and login.
- Footer includes information links (or social media links).
- Main content area for accommodation listings with placeholder for filters.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,11 +18,14 @@
...
     "@mui/icons-material": "^6.1.6",
     "@mui/material": "^6.1.6",
+    "@mui/x-date-pickers": "^7.22.2",
     "@tanstack/react-query": "^5.59.19",
+    "dayjs": "^1.11.13",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
     "react-helmet-async": "^2.0.5",
     "react-hook-form": "^7.53.1",
+    "react-icons": "^5.3.0",
```
Comment (NegatiV-64) [inline package.json]: Please use MUI Icons

File: src/modules/home/components/Filters.tsx
Comment (NegatiV-64) [inline src/modules/home/components/Filters.tsx:1]: use kebab-case for file names

File: src/modules/home/routes/home.route.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import { ApartmentCard } from '#/modules/home/components/card';
 import Box from '@mui/material/Box';
 import Typography from '@mui/material/Typography';
+import { Filters } from '../components/Filters';
 
 export function HomeRoute() {
   return (
     <Box>
+      <Filters />
```
Comment (NegatiV-64) [inline src/modules/home/routes/home.route.tsx]: I don't think that we need to have this component yet

File: package.json [hunk 3]
```diff
@@ -18,11 +18,14 @@
     "@hookform/resolvers": "^3.9.1",
     "@mui/icons-material": "^6.1.6",
     "@mui/material": "^6.1.6",
+    "@mui/x-date-pickers": "^7.22.2",
```
Comment (NegatiV-64) [inline package.json]: Why we need it?

File: src/modules/layout/components/footer/Footer.tsx
Comment (NegatiV-64) [inline src/modules/layout/components/footer/Footer.tsx:1]: use kebab-case for file names

File: src/modules/layout/components/header/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import { Logo } from './Logo';
+import { Search } from './Search';
+import { ProfileOptions } from './ProfileOptions';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+
+export function Header() {
+  const [search, setSearch] = useState(false);
+
+  function handleSearch() {
```
Comment (NegatiV-64) [inline src/modules/layout/components/header/Header.tsx]: Please remove search functionality

File: src/modules/layout/components/header/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      sx={{
+        display: 'flex',
+        flexDirection: 'row',
+        minHeight: 80,
+        borderBottom: '1px solid #ddd',
+      }}
+    >
+      <Container maxWidth="xl">
+        <Box
+          sx={{
```
Comment (NegatiV-64) [inline src/modules/layout/components/header/Header.tsx]: Move sx to separate constant `styles`

File: src/modules/layout/components/header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+export function Header() {
+  const [search, setSearch] = useState(false);
+
+  function handleSearch() {
+    setSearch((prev) => !prev);
+  }
+
+  return (
+    <Box
+      sx={{
```
Comment (NegatiV-64) [inline src/modules/layout/components/header/Header.tsx]: Move sx to separate constant `styles`

File: src/modules/layout/components/header/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          sx={{
+            display: 'flex',
+            justifyContent: { xs: 'center', md: 'space-between' },
+            alignItems: 'center',
+            minHeight: 80,
+            px: 4,
+          }}
+        >
+          <Logo />
+          <Search search={search} handleSearch={handleSearch} />
```
Comment (NegatiV-64) [inline src/modules/layout/components/header/Header.tsx]: Remove search

File: src/modules/layout/components/header/Header.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+          <ProfileOptions />
+        </Box>
+
+        <Box
+          sx={{
```
Comment (NegatiV-64) [inline src/modules/layout/components/header/Header.tsx]: Move sx to separate constant `styles`

File: src/modules/layout/components/header/Header.tsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+            gap: 2,
+            mb: 2,
+            maxHeight: search ? '100px' : '0px',
+            overflow: 'hidden',
+            opacity: search ? 1 : 0,
+            transform: search ? 'translateY(0)' : 'translateY(-10px)',
+            transition: 'max-height 0.5s ease, opacity 0.5s ease, transform 0.5s ease',
+          }}
+        >
+          <TextField sx={{ borderRadius: 20, mt: 1 }} label="Location" variant="outlined" />
```
Comment (NegatiV-64) [inline src/modules/layout/components/header/Header.tsx]: No need to have search/filter functionality

File: src/modules/layout/components/header/Header.tsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+          <LocalizationProvider dateAdapter={AdapterDayjs}>
+            <DemoContainer components={['DatePicker']}>
+              <DatePicker label="Check in" />
+            </DemoContainer>
+          </LocalizationProvider>
+
+          <LocalizationProvider dateAdapter={AdapterDayjs}>
```
Comment (NegatiV-64) [inline src/modules/layout/components/header/Header.tsx]: Same comment

File: src/modules/layout/components/header/Languages.tsx
Comment (NegatiV-64) [inline src/modules/layout/components/header/Languages.tsx:1]: We don't have functionality for multilang support

File: src/modules/layout/components/header/Languages.tsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+          horizontal: 'right',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right',
+        }}
+        sx={{ mt: 1 }}
+      >
+        {lang.map((elem) => (
+          <MenuItem key={elem.id}>
```
Comment (farrooh) [inline src/modules/layout/components/header/Languages.tsx]: All `elem.id` are `1` right now, which causes the same key error when used as keys. Maybe change them to `1, 2, 3`?

**Review by farrooh — CHANGES_REQUESTED** [review]

**Review by farrooh — COMMENTED** [review]
Also, container is not working properly on large screen. Or will it be fixed when making it responsive?

File: src/modules/layout/components/header/Header.tsx
Comment (NegatiV-64) [inline src/modules/layout/components/header/Header.tsx:1]: use kebab-case for file names

File: src/modules/layout/components/header/Logo.tsx
Comment (NegatiV-64) [inline src/modules/layout/components/header/Logo.tsx:1]: use kebab-case for file names

File: src/modules/layout/components/header/ProfileOptions.tsx
Comment (NegatiV-64) [inline src/modules/layout/components/header/ProfileOptions.tsx:1]: use kebab-case for file names

File: src/modules/layout/components/header/Search.tsx
Comment (NegatiV-64) [inline src/modules/layout/components/header/Search.tsx:1]: Let's remove search from the header

File: src/modules/layout/components/header/UserMenu.tsx
Comment (NegatiV-64) [inline src/modules/layout/components/header/UserMenu.tsx:1]: use kebab-case for file names

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]

**Review by NegatiV-64 — APPROVED** [review]

**Pull request merged**